Framework/PythonInterface/plugins/algorithms/LoadElementalAnalysisData.py

```diff
@@ -0,0 +1,177 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+
+import glob
+import os
+import numpy as np
+from mantid import config
+from mantid.api import AlgorithmFactory, AnalysisDataService, PythonAlgorithm, TextAxis, WorkspaceGroup, WorkspaceGroupProperty
+from mantid.kernel import Direction, IntBoundedValidator
+from mantid.simpleapi import ConvertToHistogram, ConvertToPointData, DeleteWorkspace, LoadAscii, WorkspaceFactory
+
+
+type_keys = {"10": "Prompt", "20": "Delayed", "99": "Total"}
+spectrum_index = {"Delayed": 1, "Prompt": 2, "Total": 3}
+num_files_per_detector = 3
+
+
+class LoadElementalAnalysisData(PythonAlgorithm):
+    def category(self):
+        """
+        Returns category
+        """
+        return "DataHandling"
+
+    def PyInit(self):
+        self.declareProperty(name='Run', defaultValue=0, validator=IntBoundedValidator(lower=0),
+                             doc='Run number to load')
+
+        self.declareProperty(WorkspaceGroupProperty(name='GroupWorkspace', defaultValue='',
+                                                    direction=Direction.Output),
+                             doc='Output group workspace for run')
+
+    def validateInputs(self):
+        issues = dict()
+        checkRun = self.search_user_dirs()
+        if not checkRun:
+            issues['Run'] = "Cannot find files for run " + self.getPropertyValue("Run")
+        return issues
+
+    def PyExec(self):
+        run = self.getPropertyValue("Run")
+        to_load = self.search_user_dirs()
+        workspaces = {
+            filename: self.get_filename(filename, run)
+            for filename in to_load if self.get_filename(filename, run) is not None
+        }
+        unique_workspaces = {}
+        for path, workspace in workspaces.items():
+            if workspace not in unique_workspaces.values():
+                unique_workspaces[path] = workspace
+        workspaces = unique_workspaces
+
+        self.format_workspace(workspaces)
+        self.merge_workspaces(run, workspaces.values())
+
+    def pad_run(self):
+        """ Pads run number: i.e. 123 -> 00123; 2695 -> 02695 """
+        return str(self.getProperty("Run").value).zfill(5)
+
+    def search_user_dirs(self):
+        """ Finds all files for the run number provided """
+        files = []
+        for user_dir in config["datasearch.directories"].split(";"):
+            path = os.path.join(user_dir, "ral{}.rooth*.dat".format(self.pad_run()))
+            files.extend([file for file in glob.iglob(path)])
+        return files
+
+    def get_filename(self, path, run):
```

Comment:
is it the path or the filename?

Reply:
Checked and it's definitely using the path so have changed to make it more consistent/understandable.

Reply:
just use the path as the variable name when you use this function (it currently uses filename)